R/utils_plot_mods.R

```diff
@@ -258,13 +258,28 @@
         ")", "-", "+", "_", "=", ";", "/", "|", "{", "}", "~"
     )[seq_len(letters.needed)]
     names(letter.labels) <- levels(as.factor(Target_data[,col.use]))
-    p <- p +
-        geom_point(
-            data=Target_data,
-            aes(x = .data[[x.by]], y = .data[[y.by]], shape = .data[[col.use]]),
-            color = "black", size=size*3/4, alpha = opacity) +
-        scale_shape_manual(
-            name = legend.title,
-            values = letter.labels)
+
+    # Determine how to add data while adding proper theming
+    aes.use <- aes(x = .data[[x.by]], y = .data[[y.by]], shape = .data[[col.use]])
+    geom.args <- list(data = Target_data, color = "black")
+
+    if (is.character(size)) {
+        aes.use <- modifyList(aes.use, aes(size = .data[[size]]))
+    } else {
+        geom.args$size <- size
+    }
```

Comment:
Lookin good!

My one note is just add back the `geom.args$size <- size*3/4` in line 269 and then I think I'm satisfied =).  It keeps the letters a bit more similar in size to the data points.  It falls away (now... thanks for the bug catch!) when the user sizes by a data column, but that can't be easily helped.

Reply:
I can handle the NEWS and version updates.  We can't successfully push to CRAN anyway until examples are trimmed down... CRAN seems to be more strict about example run times than Bioconductor.

Reply:
Alrighty, made that change.